bundles/org.openhab.binding.dali/src/main/java/org/openhab/binding/dali/internal/handler/DaliserverBridgeHandler.java

```diff
@@ -0,0 +1,237 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.dali.internal.protocol.DaliBackwardFrame;
+import org.openhab.binding.dali.internal.protocol.DaliCommandBase;
+import org.openhab.binding.dali.internal.protocol.DaliResponse;
+import org.openhab.core.common.NamedThreadFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DaliserverBridgeHandler} handles the lifecycle of daliserver connections.
+ *
+ * @author Robert Schmid - Initial contribution
+ */
+@NonNullByDefault
+public class DaliserverBridgeHandler extends BaseBridgeHandler {
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(BRIDGE_TYPE);
+
+    private final Logger logger = LoggerFactory.getLogger(DaliserverBridgeHandler.class);
+    private static final int HEARTBEAT_MINUTES = 2;
+    private static final int DALI_DEFAULT_TIMEOUT = 5000;
+
+    private DaliserverConfig config = new DaliserverConfig();
+    private @Nullable ScheduledFuture<?> heartbeat;
+    private @Nullable ExecutorService commandExecutor;
+
+    public DaliserverBridgeHandler(Bridge thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Command received on daliserver {}", command);
+
+        try {
+            // TODO: Send broadcast commands
+        } catch (DaliException e) {
+            logger.info("Error handling command: {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.info("Initializing dali bridge");
+        config = getThing().getConfiguration().as(DaliserverConfig.class);
+
+        commandExecutor = Executors.newSingleThreadExecutor(new NamedThreadFactory(thing.getUID().getAsString(), true));
+
+        if (validConfiguration(config)) {
+            // start the keepalive process
+            if (heartbeat == null) {
+                logger.info("Starting heartbeat job every {} min", HEARTBEAT_MINUTES);
+                heartbeat = this.scheduler.scheduleWithFixedDelay(this::sendHeartbeat, 0, HEARTBEAT_MINUTES,
+                        TimeUnit.MINUTES);
+            }
+        }
+    }
+
+    private boolean validConfiguration(@Nullable DaliserverConfig config) {
+        if (config == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "dali configuration missing");
+            return false;
+        }
+
+        if (config.host.isEmpty() || config.port == 0) {
```

Comment:
config.port == 0
This should be impossible as you have defined the port in the XML that it must be 1 as a minimum. The framework will enforce this.

Reply:
Thanks for your reviewing effort here and in the last time @Skinah, really appreciate it! The user could skip the check by using textual configuration.